onnxoptimizer/passes/eliminate_nop_reshape.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+#include "onnx/defs/tensor_util.h"
+#include "onnxoptimizer/pass.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+struct EliminateNopReshape final : public PredicateBasedPass {
+  explicit EliminateNopReshape()
+      : PredicateBasedPass(PassType::Nop, PassEfficiency::Complete,
+                           PassOptimizationType::Compute) {}
+
+  std::string getPassName() const override {
+    return "eliminate_nop_reshape";
+  }
+
+  bool patternMatchPredicate(Node *node) override {
+    return node->kind() == kReshape && !node->inputs()[0]->sizes().empty();
+  }
+
+  bool runTransform(Node *node, Graph &graph,
+                    NodeDestroyType &destroy_current) override {
+    const auto &data_dims = node->inputs()[0]->sizes();
```

Comment:
Please use a better name. `data_dims` is similar with another variable `shape_data` but has completely different meaning